salesagility/SuiteCRM
# Fix #5754 - Error with custom fields on getQuery from One2Many relationships

If you call `getQuery` method on One2Many relationship using custom fields on `order_by` or `where` optional parameters you've got the following error:

```
Query Failed: SELECT id FROM tasks WHERE tasks.contact_id = '62f50b22-c255-ccc9-a2e2-5a871e7b989' AND tasks.deleted=0 AND tasks_cstm.tipo_tarea_c = 'tipo1' LIMIT 0,10: MySQL error 1054: Unknown column 'tasks_cstm.tipo_tarea_c' in 'where clause'
```

This is caused due missing JOIN with cstm table.

See #5754 

## Description of changes
If optional `order_by` or `where` parameter is specified on `getQuery` method of One2Many relationship do a JOIN with custom table of module (if has custom table).





## Motivation and Context

I'm trying to get related records of a bean filtering by a custom field.

## How To Test This

1. Call `getQuery` method on One2Many relationship using custom fields on `order_by` or `where`.
2. Notice that you do not receive any result.
3. See a MySQL error on `suitecrm.log`.
4. Apply the PR and repeat steps 1, 2 and 3.
5. See that is working now.

## Types of changes


### Final checklist
